tests/test_pvc_concurrent_deletion_creation.py

```diff
@@ -0,0 +1,223 @@
+    """
+    pvc_yaml = os.path.join(TEMPLATE_DIR, "PersistentVolumeClaim.yaml")
+    pvc_parms = {'sc_name': 'test-sc', 'pvc_name': pvc_base_name}
+
+    pvc_yaml_file = templating.generate_yaml_from_jinja2_template_with_data(
+        pvc_yaml, **pvc_parms
+    )
+    pvc_obj = munchify(pvc_yaml_file)
+
+    for count in range(1, number_of_pvc + 1):
+        pvc_name = f'{pvc_base_name}{count}'
+        pvc_obj.metadata.name = pvc_name
+        with open(TEMP_YAML_FILE_PVC, 'w') as yaml_file:
+            yaml.dump(pvc_obj.toDict(), yaml_file, default_flow_style=False)
+
+        log.info(f'Creating Persistent Volume Claim {pvc_name}')
+        assert PVC.create(yaml_file=TEMP_YAML_FILE_PVC)
+        log.info(f'Created Persistent Volume Claim {pvc_name}')
+
+
+def run_async(command):
+    """
+    Run command locally and return without waiting for completion
+
+    Args:
+        command (str): The command to run on the system.
+
+    Returns:
+        An open descriptor to be used by the calling function.
+        None on error.
+    """
+    log.info(f"Executing command: {command}")
+    p = subprocess.Popen(
+        command, stdin=subprocess.PIPE, stdout=subprocess.PIPE, shell=True
+    )
+
+    def async_communicate():
+        stdout, stderr = p.communicate()
+        retcode = p.returncode
+        return retcode, stdout, stderr
+
+    p.async_communicate = async_communicate
+    return p
+
+
+def verify_pvc_bound(pvc_name):
+    """
+    Verify that the status of PVC is 'Bound'
+
+    Args:
+        pvc_name (str): Name of PVC
+
+    Returns:
+        True if PVC is in Bound state, False if PVC does not exists.
+    """
+    try:
+        pvc_info = PVC.get(pvc_name)
+        assert pvc_info['status']['phase'] == "Bound", (f'PVC {pvc_name} is '
+                                                        f'not in Bound state')
+    except exceptions.CommandFailed:
+        log.info(f"PVC {pvc_name} doesn't exist")
+        return False
+    return True
+
+
+def verify_pvc_not_exists(pvc_name):
+    """
+    Ensure that the pvc does not exists
+
+    Args:
+        pvc_name (str): Name of PVC
+
+    Returns:
+        True if PVC does not exists, False otherwise.
+    """
+    try:
+        PVC.get(pvc_name)
+        return False
+    except exceptions.CommandFailed as exp:
+        assert "not found" in str(exp)
+        log.info(f'Expected: PVC {pvc_name} does not exists ')
+    return True
+
+
+@tier1
+@pytest.mark.usefixtures(
+    test_fixture.__name__,
+)
+class TestMultiplePvcConcurrentDeletionCreation(ManageTest):
+    """
+    Test to verify concurrent creation and deletion of multiple PVCs
+    """
+    def test_multiple_pvc_concurrent_creation_deletion(self):
+        """
+        To exercise resource creation
+        """
+        sc_parms = {'sc_name': 'test-sc'}
+        pvc_base_name = 'test-pvc'
+        number_of_pvc = 500
+
+        # Create storage class
+        create_storage_class(**sc_parms)
+
```

Comment:
@jilju are we creating a separate project/namespace for new pvcs. Its always better to have a new project rather than using default/openshift-storage.

Reply:
Done